followthemoney/graph.py

```diff
@@ -11,26 +16,31 @@ class Node(object):
     address or phone number."""
     __slots__ = ['type', 'value', 'id', 'proxy', 'schema']
 
-    def __init__(self, type_, value, proxy=None, schema=None):
-        self.type = type_
-        self.value = value
-        self.id = type_.node_id_safe(value)
-        self.proxy = proxy
-        self.schema = schema if proxy is None else proxy.schema
+    def __init__(self, type_: PropertyType, value: str,
+                 proxy: Optional[EntityProxy]=None,
+                 schema: Optional[Schema]=None):
+        self.type: PropertyType = type_
+        self.value: Optional[str] = value
```

Comment:
Is value optional? That would then be when we get into the whole optional node ID mess.

Reply:
It doesn't have to, but then `EntityType.caption` has to return a string, and not `None`: https://github.com/alephdata/followthemoney/blob/91e56b450a9c1bc29221e40cfc2512dfd1f97dfd/followthemoney/types/entity.py#L32